### ISSUE TITLE
Chapter 3 - Using Control.Plus without installing it

### ISSUE THREAD

**Issue by JulianLeviston** [paf31/purescript-book#91]
In Ch3, `Control.Plus` is used without it being imported. Should add `bower install purescript-control --save` and make a note about selecting a correct version of the prelude if not specifying specific versions.

https://github.com/paf31/purescript-book/blame/master/text/chapter3.md#L27

**Comment by JulianLeviston** [comment]
Also, when that is installed, `pulp build` doesn't then work. (Bower 1.7.9, psc 0.8.2, pulp 8.1.0, macOS 10.12.5)

**Comment by JulianLeviston** [comment]
This is the error message:

```
* Building project in /Users/julian/code/purescript/address-book
psc: No files found using pattern: src/**/*.js
Error 1 of 3:

  at /Users/julian/code/purescript/address-book/bower_components/purescript-prelude/src/Data/Unit.purs line 11, column 29 - line 11, column 29

    Unable to parse module:
    unexpected "Type"
    expecting kind


  See https://github.com/purescript/purescript/wiki/Error-Code-ErrorParsingModule for more information,
  or to contribute content related to this error.

Error 2 of 3:

  at /Users/julian/code/purescript/address-book/bower_components/purescript-prelude/src/Data/NaturalTransformation.purs line 19, column 10 - line 19, column 10

    Unable to parse module:
    unexpected "type"
    expecting qualifier, identifier, proper name or symbol


  See https://github.com/purescript/purescript/wiki/Error-Code-ErrorParsingModule for more information,
  or to contribute content related to this error.

Error 3 of 3:

  at /Users/julian/code/purescript/address-book/bower_components/purescript-prelude/src/Prelude.purs line 46, column 36 - line 46, column 36

    Unable to parse module:
    unexpected "type"
    expecting identifier, (, proper name, "class", "module" or )


  See https://github.com/purescript/purescript/wiki/Error-Code-ErrorParsingModule for more information,
  or to contribute content related to this error.

* ERROR: Subcommand terminated with exit code 1
```

**Comment by paf31** [comment]
You need to use the 0.10.x series of the compiler with the current version of the book. Also, the introduction to chapter 3 notes that you need to run `bower update` to pull in dependencies.

**Comment by JulianLeviston** [comment]
Might be worthwhile mentioning that somewhere, unless it's just a case that I missed it.

Also, I updated to version 11, and it seems to now not work because it doesn't know where to find Data.Maybe. I'll go back to where I was up to in the tutorial and give it another shot.

Ordinarily at this point I'd just go and work out how the schema for discovering module names works at this point, but I'm by no means a newbie, and I wanted to give you feedback as it seemed appropriate as it's introductory material, to save people from issues. :man_shrugging: sorry if the feedback is not wanted.

**Comment by paf31** [comment]
>  because it doesn't know where to find Data.Maybe

Can you please paste the error message?

**Comment by JulianLeviston** [comment]
I should really run through it again before I do, but:

```
➜  address-book  pulp build
* Building project in /Users/julian/code/purescript/address-book
Error 1 of 2:

  in module AddressBook
  at src/Data/AddressBook.purs line 6, column 1 - line 6, column 42

    Module Data.List was not found.
    Make sure the source file exists, and that it has been provided as an input to psc.


  See https://github.com/purescript/documentation/blob/master/errors/ModuleNotFound.md for more information,
  or to contribute content related to this error.

Error 2 of 2:

  in module AddressBook
  at src/Data/AddressBook.purs line 7, column 1 - line 7, column 26

    Module Data.Maybe was not found.
    Make sure the source file exists, and that it has been provided as an input to psc.


  See https://github.com/purescript/documentation/blob/master/errors/ModuleNotFound.md for more information,
  or to contribute content related to this error.


* ERROR: Subcommand terminated with exit code 1
```

**Comment by paf31** [comment]
Have you run `bower update`?

**Comment by JulianLeviston** [comment]
Not sure. Nothing in the book told me to. I guess I should. (Doing so now). Did that. No difference in output.

**Comment by JulianLeviston** [comment]
(BTW, I hope you don't misunderstand my intent here — not overly concerned with my particular case, I'm sure I can easily fix it, more concerned that people might be using the PS book and getting stuck)

**Comment by paf31** [comment]
The start of chapter 3 says this:

> Assuming you have cloned the book's source code repository, the project for this chapter can be built using Pulp, with the following commands:
> 
> ```
> $ cd chapter3
> $ bower update
> $ pulp build
> ```

**Comment by paf31** [comment]
Please try this - clean your Bower cache (`rm -rf bower_components && bower cache clean`) and output directory (`rm -rf output`), and then run those steps (`bower update; pulp build`) again.

**Comment by JulianLeviston** [comment]
Right. I didn't clone the book's source code. Funny, I was just thinking about that. :) Mind you, as a user I'd like to know how to install and use packages without using a curated repo, but I'm sure you get to that later in the book so I guess I should just follow the happy path if I've got my noob hat on :)

**Comment by JulianLeviston** [comment]
(BTW same issue, but seriously don't worry I'll fix it myself)

**Comment by JulianLeviston** [comment]
Also, I should say, massive congrats on the lang. It's freaking awesome to be able to have a haskell-like in the browser! :)

**Comment by JulianLeviston** [comment]
Okay, so FWIW, what I didn't know was that I had to install packages for `Data.List` and `Data.Maybe` to work, so of course, this is what fixed my issues:

```
bower install purescript-maybe --save-dev
bower install purescript-lists --save-dev
```

I worked this out by going to pursuit, typing `maybe` (A function I happen to know exists in `Data.Maybe`, because I know Haskell reasonably well), which showed the name of the package I was missing, and then `(:)` because I assumed that'd exist in a list package. It turns out it's called `purescript-lists` which isn't what I'd expect, but nonetheless - this is an issue I found often a problem when initially learning Haskell, too... never knowing where to install things from, so I'm happy to find that pursuit is much better for this! ❤️ thank you.
